### ISSUE TITLE
[GitLab] PR Decorating not working

### ISSUE THREAD

**Issue by tienhm0202** [mc1arke/sonarqube-community-branch-plugin#247]
**Describe the bug**
- PR can be created in SonarQube server but can not decorate to Gitlab Merge Request.

**To Reproduce**

1. GitlabCI:

```
./gradlew sonarqube \
    -Dsonar.host.url=$SONAR_HOST_URL \
    -Dsonar.qualitygate.wait=true \
    -Dsonar.pullrequest.branch=${CI_MERGE_REQUEST_SOURCE_BRANCH_NAME} \
    -Dsonar.pullrequest.base=${CI_MERGE_REQUEST_TARGET_BRANCH_NAME} \
    -Dsonar.pullrequest.key=${CI_MERGE_REQUEST_ID} \
    -Dsonar.gitlab.project_id=${CI_MERGE_REQUEST_PROJECT_PATH} \
    -Dsonar.gitlab.commit_sha=${CI_MERGE_REQUEST_SOURCE_BRANCH_SHA} \
    -Dsonar.gitlab.ref_name=${CI_COMMIT_REF_NAME}
```

2. Check SonarQube server

- Go to Project Settings > General Settings > Pull Request Decorate

![PR Decorate](https://i.imgur.com/nv46u66.png)
![ALM Settings](https://i.imgur.com/Tvn1rSk.png)

**Expected behavior**
Can decorate Gitlab Merge Request

**Software Versions**
 - SonarQube Version: 8.4-community
 - Plugin Version: 1.4.0
 - Gradle: 6.1.1
 - Gradle-sonarqube-plugin: 3.0
 - Gitlab: 11.11

**Comment by PiekJ** [comment]
Plugin Version 1.4.0 does not work with SonarQube version 8.2+. The master branch should work. Please build the project yourself and try again.

**Comment by tienhm0202** [comment]
@PiekJ After build master branch, but I can't get merge request decorator.

I can get merge request decorator config in project

![](https://i.imgur.com/7LVZPpy.png)

But there is still nothing in Gitlab.

**Comment by PiekJ** [comment]
In order to decorate your pull request you have to configure some parameters for your scanner. Please read this documentation on how to setup your Sonar project: https://docs.sonarqube.org/8.4/analysis/pr-decoration/ and this on how to configure the scanner: https://docs.sonarqube.org/latest/analysis/pull-request/
Note that if you use these parameters, you must not use the `sonar.branch.name` property.

**Comment by tienhm0202** [comment]
@PiekJ Yes yes.

- About setup Sonar: I set up ALM as documents
- About scanner:

```
./gradlew sonarqube \
    -Dsonar.host.url=$SONAR_HOST_URL \
    -Dsonar.qualitygate.wait=true \
    -Dsonar.pullrequest.branch=${CI_MERGE_REQUEST_SOURCE_BRANCH_NAME} \
    -Dsonar.pullrequest.base=${CI_MERGE_REQUEST_TARGET_BRANCH_NAME} \
    -Dsonar.pullrequest.key=${CI_MERGE_REQUEST_ID} \
    -Dsonar.gitlab.project_id=${CI_MERGE_REQUEST_PROJECT_PATH} \
    -Dsonar.gitlab.commit_sha=${CI_MERGE_REQUEST_SOURCE_BRANCH_SHA} \
    -Dsonar.gitlab.ref_name=${CI_COMMIT_REF_NAME}
```
As my first comment https://github.com/mc1arke/sonarqube-community-branch-plugin/issues/247#issue-710106201

**Comment by eugentius** [comment]
@tienhm0202 , try "-Dsonar.pullrequest.gitlab.projectId=" instead of "-Dsonar.gitlab.project_id"

**Comment by truongsinh** [comment]
@tienhm0202 `CI_MERGE_REQUEST_IID` instead of `CI_MERGE_REQUEST_ID`, or just leave it blank all together.
```
./gradlew sonarqube \
    -Dsonar.host.url=$SONAR_HOST_URL \
    -Dsonar.qualitygate.wait=true \
```

The problems:
1. Gitlab expect "local id" of the MR
2. sonarqube cli auto detects those env vars
3. `sonar.gitlab.*` is from a defunc community plugin

**Comment by felipegustavo** [comment]
The bug cited in this issue seems to be resolved in version 1.5.0. The merge request decoration worked just fine.

I used the parameters as suggest by @truongsinh in the comment above

**Software Versions**
- SonarQube 8.4-community
- Maven 3.36
- GitLab 13.11
- SonarQube Scanner For Maven 3.7
